Tools/Scripts/libraries/webkitapipy/webkitapipy/allow.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (C) 2025 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import sys
+from dataclasses import dataclass, field
+from pathlib import Path
+from enum import Enum
+from typing import Any, Optional
+
+if sys.version_info < (3, 11):
+    from pip._vendor import tomli
```

Comment:
We have no guarantee that `pip` is installed, nor what version of `pip` is installed, so we absolutely cannot rely on its vendoring. If we have dependencies, we need to install them ourselves (likely via the autoinstaller).

Reply:
Hm… it's guaranteed to be present in the Xcode toolchain, is that not enough? The core `audit-spi` workflow needs to function without network access, so that precludes autoinstalling `tomli`. We could vendor tomli ourselves, but I think that's overkill when this hack _does_ work on the environments we need to support.

Reply:
One thing we could do is declare a real dependency on `tomli`, try to faithfully import it, possibly even autoinstall it, and fallback to pip's vendored copy as a last-ditch effort. That would keep this functional on internal systems with no network access, but not expose other clients to this hack. Would that be sufficient?

Reply:
Per our discussion, I've vendored tomli in `webkitapipy._vendor.tomli`.